### DISCUSSION TITLE
With autotrigger enabled, accept suggestion and move to a new line?

### DISCUSSION THREAD

**Discussion by raine** [zbirenbaum/copilot.lua#109] [Q&A] [answered]
With auto trigger enabled, it would seem natural to be able to move to a new line after accepting a suggestion, because it won't suggest a new line until enter a newline manually. So you have to do `<M-l><CR><M-l><CR>...` to autocomplete multiple lines. Or how do people use this for that use case, through `:Copilot panel`?

**Comment by MunifTanjim** [comment, marked as answer]
You can make your custom accept keymap for that:

```lua
require("copilot").setup({
  -- ...
  suggestion = { 
    keymap = {
      accept = false, -- disable default accept keymap
      -- ...
    },
    -- ...
  },
  -- ...
})

vim.keymap.set("i", "<M-l>", function ()
  require("copilot.suggestion").accept()

  -- extra newline
  vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes("<CR>", true, false, true), "n", false)
end, {
  desc = "[copilot] accept suggestion",
  silent = true,
})


```

**Reply by raine** [reply]
Munif, thank you for the example. I need to learn brush up my lua and learn the nvim apis someday. :)

I did a minor change to also make it give the next suggestion after `<CR>`. That way I don't have keep auto suggest enabled. Seems to work quite well. I think a mapping that does something like this might be valuable as built-in in copilot.lua.

```lua
-- https://github.com/zbirenbaum/copilot.lua/discussions/109#discussioncomment-4760925
vim.keymap.set("i", "<M-l>", function ()
  require("copilot.suggestion").accept()

  -- extra newline
  vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes("<CR>", true, false, true), "n", false)
  require("copilot.suggestion").next() -- <-- this
end, {
  desc = "[copilot] accept suggestion",
  silent = true,
})
```
